### DISCUSSION TITLE
[bug] Storage Product deletion

### DISCUSSION THREAD

**Discussion by waran70** [YetiForceCompany/YetiForceCRM#16328] [General]
Bug or design?

Product can be permanently deleted from own module as well as from Storage. 
For me it smells like design error. Am I right or wrong?

**Comment by davide-alghi** [comment]
Hi @waran70,
it depends what you mean.
Many related systems allow different deleting levels: from "logical" to "physical".
It's certain that the "physical" deleting, definitively breaks all possible relationships, but it must be taken into account that some operations, such as deleting and many configurations, must be assigned only to people who are competent and who know what they are doing.
Perhaps I would add, in the profiles configuration, the distinction between "logical" and "physical" deleting privilege.
Yes, deleting can lead to errors and that is why certain features need to be assigned to the right profiles.
It is always better to have a feature available and not use it, rather than not have it.
The front-end user needs to have deleting privileges on only a few modules, such as the calendar. If she/he has to delete a record, a request must be sent to the super-admin who will do the necessary checks.

This is my opinion.

**Comment by waran70** [comment]
This is what I mean. The deletion of products from the Storage module is too confusing for users. 
![image](https://user-images.githubusercontent.com/6852348/98186096-9ac67f00-1f40-11eb-9443-867f984c7382.png)
